### ISSUE TITLE
`Failed to store chunk` errors with CCA and Fabric API

### ISSUE THREAD

**Issue by dicedpixels** [Ladysnake/Cardinal-Components-API#188]
**Test plan:**

Launch an instance:
- `fabricloader 0.16.9`
- `minecraft 1.21.3`
- `cardinal-components 6.2.0`
- `fabric-api 0.107.0+1.21.3`

Create a single-player world.

Observe the endless `Failed to store chunk` errors.

Log: [iIMwwMC.txt](https://github.com/user-attachments/files/17679022/iIMwwMC.txt) - [mclogs](https://mclo.gs/iIMwwMC)

**Comment by demkom58** [comment]
I experienced same thing on Dedicated server, here is deobfuscated stacktrace:
```
 [IO-Worker-7/ERROR] (Minecraft) Failed to store chunk [3, -3]
 java.lang.NullPointerException: Cannot invoke "net.minecraft.nbt.Tag.getId()" because "tag" is null
	at knot/net.minecraft.nbt.CompoundTag.writeNamedTag(CompoundTag.java:496) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.nbt.CompoundTag.write(CompoundTag.java:176) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.nbt.NbtIo.writeUnnamedTag(NbtIo.java:368) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.nbt.NbtIo.writeUnnamedTagWithFallback(NbtIo.java:373) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.nbt.NbtIo.write(NbtIo.java:327) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.world.level.chunk.storage.RegionFileStorage.write(RegionFileStorage.java:122) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.world.level.chunk.storage.IOWorker.runStore(IOWorker.java:227) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.world.level.chunk.storage.IOWorker.storePendingChunk(IOWorker.java:216) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.util.thread.StrictQueue$RunnableWithPriority.run(StrictQueue.java:99) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.Util.runNamed(Util.java:287) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.util.thread.AbstractConsecutiveExecutor.pollTask(AbstractConsecutiveExecutor.java:47) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at knot/net.minecraft.util.thread.AbstractConsecutiveExecutor.run(AbstractConsecutiveExecutor.java:55) ~[minecraft-common-7e133586f8-1.21.3-loom.mappings.1_21_3.layered+hash.2198-v2.jar:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

**Comment by boyonk913** [comment]
To add to this: for me this only happened when CCA was on its own mod. When I included it in my own mod, the problem ceased to exist.

**Comment by demkom58** [comment]
> To add to this: for me this only happened when CCA was on its own mod. When I included it in my own mod, the problem ceased to exist.

I haven't tried to use packaged jar with included dependency, but with just `runServer` gradle task it still happens.

**Comment by Pyrofab** [comment]
Apologies, I had somehow missed this initial report. It has now been fixed in 6.2.1. Also, thank you for the reproduction plan, turns out our test mod was actually preventing the issue from popping up.
